src/analyze.rs

```diff
@@ -42,6 +40,9 @@ pub struct Info<'a> {
     /// previous character was. E.g. `^` matches if there's no previous
     /// character; `(?m:^)` matches if the previous character was a newline.
     pub looks_left: bool,
+
+    pub expr: &'a Expr,
+    pub children: Vec<Info<'a>>,
```

Comment:
This is unrelated, but makes the "analyze" command output easier to read.

Reply:
Agree.